TheWarWithin/MageArcane.lua

```diff
@@ -761,25 +761,112 @@ end )
 local totm_casts = 0
 local clearcasting_consumed = 0
 
+-- Intuition Bad Luck Protection tracking
+local intuitionBLPStacks = 0
+local intuition_blp_last_er_ae_timestamp = 0
+local intuition_blp_has_clearcasting = false
+local intuition_blp_is_harmony_intuition = false
+
+-- Intuition consumption tracking for charge flicker fix
+local intuition_just_consumed = false
+local intuition_consumed_timestamp = 0
+
+-- Spell IDs that can trigger BLP stack increment
+local intuitionBLPSpellIDs = {
+    [30451] = true,   -- arcane_blast
+    [5143] = true,    -- arcane_missiles
+    [44425] = true,   -- arcane_barrage
+    [1449] = true,    -- arcane_explosion
+    [365350] = true,  -- arcane_surge
+}
+
 spec:RegisterHook( "COMBAT_LOG_EVENT_UNFILTERED", function( _, subtype, _, sourceGUID, sourceName, _, _, destGUID, destName, destFlags, _, spellID, spellName )
     if sourceGUID == GUID then
+        local now = GetTime()
+
         if subtype == "SPELL_CAST_SUCCESS" and spellID == 321507 then
             totm_casts = ( totm_casts + 1 ) % 2
 
         elseif ( subtype == "SPELL_AURA_APPLIED" or subtype == "SPELL_AURA_REFRESH" ) and spellID == 458411 then
             -- Handle Arcane Orb projectile.
             local travel = ( UnitExists( "target" ) and select( 2, RC:GetRange( "target" ) ) or 40 ) / 30
-            state:QueueEvent( "arcane_orb", GetTime(), 0.05 + travel, "PROJECTILE_IMPACT", UnitGUID( "target" ), true )
+            state:QueueEvent( "arcane_orb", now, 0.05 + travel, "PROJECTILE_IMPACT", UnitGUID( "target" ), true )
 
         elseif subtype == "SPELL_AURA_REMOVED" and ( spellID == 276743 or spellID == 263725 ) then
             -- Clearcasting was consumed.
-            clearcasting_consumed = GetTime()
+            clearcasting_consumed = now
+
+        -- Intuition BLP tracking
+        elseif state.talent.intuition.enabled then
+            local has_intuition_buff = GetPlayerAuraBySpellID( 1223797 ) -- intuition buff
```

Comment:
I'm going to leave this alone for now, but it looks like there's opportunity for efficiency improvements; we're potentially firing this API function for thousands of combat log subevents that don't need it at all.  Since you're already watching for `SPELL_AURA_APPLIED` and `SPELL_AURA_REMOVED`, we could track that Intuition is up/down with a variable instead of going to the API every time.